server/pkg/store/astra/db.go

```diff
@@ -308,3 +310,291 @@ func (a *AstraServerStore) getClusterID(orgID, clusterName string) (string, erro
 
 	return "", fmt.Errorf("cluster not found")
 }
+
+func (a *AstraServerStore) isAppExistsInStore(name, version string) (bool, error) {
+	selectClusterQuery := &pb.Query{
+		Cql: fmt.Sprintf("Select name, version FROM %s.app_config WHERE name='%s' AND version ='%s';",
+			a.keyspace, name, version),
+	}
+
+	response, err := a.c.Session().ExecuteQuery(selectClusterQuery)
+	if err != nil {
+		return false, fmt.Errorf("failed to initialise db: %w", err)
+	}
+
+	result := response.GetResultSet()
+	if len(result.Rows) > 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (a *AstraServerStore) AddAppToStore(config *types.StoreAppConfig) error {
+	appExists, err := a.isAppExistsInStore(config.AppName, config.Version)
+	if err != nil {
+		return fmt.Errorf("failed to store app config : %w", err)
+	}
+
+	if appExists {
+		return fmt.Errorf("app is already available")
+	}
+
```

Comment:
this check not needed.
insert will fail with app exist would be enough.

For synchronize app case 
addorupdateTostore needed, that would do upsert query like below
INSERT INTO your_table (id, column1, column2)
        VALUES (?, ?, ?)
        IF NOT EXISTS